core/src/debug_ui/domain.rs

```diff
@@ -0,0 +1,88 @@
+use egui::{collapsing_header::CollapsingState, TextEdit, Ui, Window};
+
+use crate::{avm2::Domain, context::UpdateContext};
+
+use super::{handle::DomainHandle, Message};
+
+#[derive(Debug, Default)]
+pub struct DomainListWindow {
+    search: String,
+}
+
+impl DomainListWindow {
+    pub fn show(
+        &mut self,
+        egui_ctx: &egui::Context,
+        context: &mut UpdateContext,
+        messages: &mut Vec<Message>,
+    ) -> bool {
+        let mut keep_open = true;
+
+        Window::new("Domain List")
+            .open(&mut keep_open)
+            .show(egui_ctx, |ui| {
+                TextEdit::singleline(&mut self.search)
+                    .hint_text("Search")
+                    .show(ui);
+                ui.add_space(10.0);
+                // Let's search ascii-insensitive for QOL
+                let search = self.search.to_ascii_lowercase();
+                let domain = context.avm2.playerglobals_domain();
+                egui::ScrollArea::both().show(ui, |ui| {
+                    self.show_domain(ui, context, domain, messages, &search)
+                });
+            });
+        keep_open
+    }
+
+    #[allow(clippy::only_used_in_recursion)]
+    pub fn show_domain<'gc>(
```

Comment:
Is this `allow` still really necessary? It's clearly also used non-recursively just a few lines above. Does Clippy not detect that - and if so, is this a Clippy bug? Or am I missing something?

Reply:
This is for a parameter, not for the function itself

Reply:
Oh, it's about `self`. If you think it will be needed for when the `Class` buttons get wired up, it's fine to `allow` it - otherwise, it might be better to remove the `&mut self,`, and call as `Self::show_domain` instead, as Clippy suggests.
(Yeah I know it's nitpicking at this point, but maintaining code quality is finicky like this.)

Reply:
Yeah, I'm going to need to store some data when I implement the `Class` viewer